crates/primitives/src/transaction/error.rs

```diff
@@ -2,79 +2,85 @@ use crate::{GotExpectedBoxed, U256};
 
 /// Represents error variants that can happen when trying to validate a
 /// [Transaction](crate::Transaction)
-#[derive(Debug, Clone, Eq, PartialEq, thiserror_no_std::Error)]
+#[derive(Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
 pub enum InvalidTransactionError {
     /// The sender does not have enough funds to cover the transaction fees
-    #[error(
+    #[cfg_attr(feature = "std", error(
```

Comment:
Since we feature gated the entire enum do we have to do it for every case again?

Reply:
Actually I am rebasing on latest main from reth, maybe it won't be needed!

Thank you for flagging